ten-simple-rules-dockerfiles.Rmd

```diff
@@ -109,22 +113,36 @@ In fact, the `Dockerfile` itself can be a powerful documentation to show from wh
 # 1. Consider tools to assist with Dockerfile generation {-}
 \rulelabel{1}{rule:tools}
 
-Writing a `Dockerfile` from scratch is not that simple, and even experts sometimes take shortcuts.
-Thus, it is a good strategy to first look to tools that can help to generate a `Dockerfile` for you.
-Such tools have likely thought about and implemented good practices, and they may have added newer practices when reapplied at a later point in time.
-Therefore the most important rule is to apply a multi-step process for your specific use case.
-First, you want to determine if there is an already existing container that you can use, and in this case, use it and add to your workflow documentation instructions for doing so.
-As an example, you might be doing some kind of interactive development.
-For interactive development environments such as notebooks and development servers or databases, you can readily find containers that come installed with all the software that you need.
-Second, in the case that there isn't an existing container for your needs, you might next look to well-maintained tools to help with `Dockerfile` generation.
-These tools can add required software packages to an existing image without you having to manually write a `Dockerfile` at all.
-Well-maintained includes the tool's own stability and usability, but also that suitable base images are used, likely from the official Docker library [@docker_inc_official_2019], to ensure that the container has the most recent security fixes for the operating system in question.
-Third, you may want to write you own `Dockerfile` from scratch if these tools do not suffice for your needs.
-_In this case, follow the remaining rules._
+Rule 1 could informally be described as "Don't bother to write a
+Dockerfile!"  Writing a `Dockerfile` from scratch can be difficult
+and even experts sometimes take shortcuts.  Thus, it is a good
+strategy to first look to tools that can help to generate a
+`Dockerfile` for you.  Such tools have likely thought about and
+implemented good practices, and they may have added newer practices
+when reapplied at a later point in time.  Therefore the most important
+rule is to apply a multi-step process for your specific use case.
+First, you want to determine if there is an already existing container
+that you can use, and in this case, use it and add to your workflow
+documentation instructions for doing so.  As an example, you might be
+doing some kind of interactive development.  For interactive
+development environments such as notebooks and development servers or
+databases, you can readily find containers [SJE: where do you look to find containers?] that come installed with
+all the software that you need.  Second, in the case that there isn't
+an existing container for your needs, you might next look to
+well-maintained tools to help with `Dockerfile` generation.  These
```

Comment:
This is a good question and mostly a mess - I think people do Google searches and wind up at container registries, and the registries take them to GitHub somewhere to inspect the container recipe. Often the container recipes aren't provided and they use the base verbatim anyway.